src/test/kotlin/com/openlattice/data/LocalAwsBlobDataServiceTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.openlattice.data
+
+import com.openlattice.ResourceConfigurationLoader
+import com.openlattice.data.storage.ByteBlobDataManager
+import com.openlattice.data.storage.LocalAwsBlobDataService
+import com.openlattice.datastore.configuration.DatastoreConfiguration
+import org.junit.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.net.URL
+import java.util.*
+
+class LocalAwsBlobDataServiceTest {
+    private val logger: Logger = LoggerFactory.getLogger(LocalAwsBlobDataServiceTest::class.java)
+
+    companion object {
+        @JvmStatic
+        private lateinit var byteBlobDataManager: ByteBlobDataManager
+        private var key1 = ""
+
+        @BeforeClass
+        @JvmStatic
+        fun setUp() {
+            val config = ResourceConfigurationLoader.loadConfiguration(DatastoreConfiguration::class.java)
+            val byteBlobDataManager = LocalAwsBlobDataService(config)
+            this.byteBlobDataManager = byteBlobDataManager
+        }
+
+        @AfterClass
+        @JvmStatic
+        fun cleanUp() {
+            byteBlobDataManager.deleteObject(this.key1)
+        }
+
+    }
+
+    @Test
+    fun testPutAndGetObject() {
+        val data = ByteArray(10)
+        Random().nextBytes(data)
+        for (i in 1..3) {
+            key1 = key1.plus(UUID.randomUUID().toString())
+        }
+        key1 = key1.plus(data.hashCode())
```

Comment:
You will have to use Hasher or check out the PostgresDataHasher class, which has reasonable implementations for the EdmPrimitiveTypeKinds.

Reply:
I implemented this differently but am not sure it is what you are looking for. Also changed in PostgresEntityDataQueryService